pumps/prometheus.go

```diff
@@ -178,12 +174,39 @@ func (p *PrometheusPump) Init(conf interface{}) error {
 	go func() {
 		log.Fatal(http.ListenAndServe(p.conf.Addr, nil))
 	}()
-	p.log.Info(p.GetName() + " Initialized")
+	p.log.WithField("disabled_metrics", disabledMetrics).Info(p.GetName() + " Initialized")
 
 	return nil
 }
 
-//InitCustomMetrics initialise custom prometheus metrics based on p.conf.CustomMetrics and add them into p.allMetrics
+func (p *PrometheusPump) initBaseMetrics() []string {
+	for _, metric := range p.allMetrics {
+		metric.aggregatedObservations = p.conf.AggregateObservations
+		errInit := metric.InitVec()
```

Comment:
@hsyed-dojo I think it's going to be better to create the `toDisableSet` first, and check if we should `errInit := metric.InitVec()` based on the metric name here. In this way, we avoid initializing the metric if it's already disabled and avoid the need of unregistering it from Prometheus client. 
Maybe we can create a `PrometheusMetric` method for this and make this cleaner / easier to test.   
What do you think?

Reply:
@tbuchaillot that is the approach I would have taken if the data flow did not already have the concept of disabling a metric during exposition. Having two codepaths to turn off metric exposition seems less clear to reason about.

Happy to update. Let me know.

Reply:
@hsyed-dojo the current concept of disabling metrics is only for initialization failure. In fact, we are enabling when init correctly instead of disabling them IIRC.
If you want to give it a try to this approach, it would be amazing :)

Reply:
@tbuchaillot done